### PR TITLE
chore: Remove hooks folder

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR removes the hooks folder. 

## How does it work?
The main idea of the `sdk` folder is to contain hooks with business logic that are used across the app. Turns out a new folder `hooks` was created, which lead to duplication of purposes on the filesystem. 
This PR unifies both hooks and sdk folders. Initially, I thought about moving the files "as is" from hooks to sdk, however, a closer inspection revealed that hooks on the `hooks` folder were only being used in a single file. I then, decided to inline these functions with the intent of having a smaller code base.

## How to test it?
Everything should work as before

## Checklist
- [x] CHANGELOG entry added

### CODE REVIEW

File: src/components/common/PostalCode/PostalCodeInput.tsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
+const POSTAL_CODE_STORAGE_KEY = 'main::store::postalCode'
 const POSTAL_CODE_INPUT_ID = 'postal-code-input'
 
 export default function PostalCodeInput() {
-  const [localPostalCode, setLocalPostalCode] = useState<string>('')
-  const [, setPostalCode] = usePostalCode()
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setLocalPostalCode(event.target.value)
-  }
+  const [code, setPostalCode] = useStorage<string>(POSTAL_CODE_STORAGE_KEY, '')
```
Comment (filipewl) [inline src/components/common/PostalCode/PostalCodeInput.tsx:12]: Just to better indicate that these variables are related, what do you think of renaming `code` to `postalCode` or `setPostalCode` to `setCode`?

**Review by filipewl — APPROVED** [review]
LGTM! I remember a previous comment where you mentioned this need to move everything from `/hook` to `/sdk`.

**Review by eduardoformiga — APPROVED** [review]
LGTM! Moving the `useSearchParams.ts` inline looks harder for readability. But since we just use it on one file and it's good to improve the codebase size, I agree with this change :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - useWindowDimensions hook
 
 ### Removed
-
+- Hooks folder
```
Comment (igorbrasileiro) [inline CHANGELOG.md:39]: I think we could say "removing hooks folder and migrating these hooks to sdk ou inline them on components"

File: src/components/common/PostalCode/PostalCodeInput.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply src/components/common/PostalCode/PostalCodeInput.tsx:12]: This hook was planned to be used in other components/hooks. But we was doing the feature step by step

File: src/components/common/PostalCode/PostalCodeInput.tsx [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
+  const ref = useRef<HTMLInputElement>(null)
 
   const handleSubmit = (event: KeyboardEvent<HTMLInputElement>) => {
-    if (event.key === 'Enter' && Boolean(localPostalCode)) {
-      setPostalCode(localPostalCode)
+    const val = ref.current?.value
```
Comment (victorhmp) [inline src/components/common/PostalCode/PostalCodeInput.tsx:16]:
```suggestion
    const value = ref.current?.value
```

or 

```
const currentValue = ref.current?.value
```

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Review by victorhmp — APPROVED** [review]

**Pull request merged**